src/components/app-configure/SettingCard.vue

```diff
@@ -1,15 +1,48 @@
 <script lang="ts" setup>
+import { computed, type ComputedRef } from 'vue'
+
 import VCard from '@/components/lib/VCard/VCard.vue'
 import VSeperator from '@/components/lib/VSeperator/VSeperator.vue'
 import VStack from '@/components/lib/VStack/VStack.vue'
+import VSwitch from '@/components/lib/VSwitch/VSwitch.vue'
+
+interface SettingsProps {
+  hasHeaderSwitch?: boolean
+  switchState?: boolean
+  isSwitchDisabled?: boolean
+}
+
+const props = withDefaults(defineProps<SettingsProps>(), {
+  hasHeaderSwitch: false,
+  switchState: false,
+  isSwitchDisabled: false,
+})
+
+const emit = defineEmits(['switchStateChange'])
+
+const headerSwitchState: ComputedRef<boolean> = computed(
+  () => props.switchState
+)
+
+function handleSwitchStateChange(value: boolean) {
+  emit('switchStateChange', value)
+}
 </script>
 
 <template>
   <VCard variant="elevated" class="settings-card">
     <VStack direction="column" gap="2rem" class="flex-grow">
-      <h3 class="text-uppercase">
-        <slot name="title" />
-      </h3>
+      <VStack justify="space-between">
+        <h3 class="text-uppercase">
+          <slot name="title" />
+        </h3>
+        <VSwitch
+          v-if="props.hasHeaderSwitch"
+          :disabled="isSwitchDisabled"
+          :model-value="headerSwitchState"
+          @update:model-value="handleSwitchStateChange"
+        />
```

Comment:
Prefer using a `slot` here instead. I'll have more flexibility in what content goes on the right-handle side of the card title, and we won't need to add props or emit events from `SettingCard` only to pass them to `VSwitch`.

```
<VStack justify="space-between">
  <h3 class="text-uppercase">
    <slot name="title" />
  </h3>
  <slot name="controls" />
</VStack>
```

Usage:

```
<SettingCard>
  <template #title>Aggregate Login</template>
  <template #controls>
    <VSwitch
      :disabled="isAppConfigured"
      :model-value="hasAggregateLogin"
      @update:model-value="handleSwitchStateChange"
    />
  </template>
  <template #description
    >Allow your users to authenticate themselves by signing transactions
    using wallets. <a>LEARN MORE</a></template
  >
</SettingCard>
```